src/shogun/lib/SGMatrix.cpp

```diff
@@ -409,14 +422,23 @@ SGVector<T> SGMatrix<T>::get_column(index_t col) const
 }
 
 template <class T>
-void SGMatrix<T>::set_column(index_t col, const SGVector<T> vec)
+void SGMatrix<T>::set_column(index_t col, const SGVector<T>& vec)
 {
 	assert_on_cpu();
 	ASSERT(!vec.on_gpu())
 	ASSERT(vec.vlen == num_rows)
 	sg_memcpy(&matrix[num_rows * col], vec.vector, sizeof(T) * num_rows);
 }
 
+template <class T>
+void SGMatrix<T>::set_column(index_t col, SGVector<T>&& vec)
```

Comment:
does exactly the same thing as above, but users can still pass xvalues (or is it prvalues)? example:
```cpp
mymatrix.set_column(idx, linalg::some_op(...))
```